### DISCUSSION TITLE
ResourceType : Bundle

### DISCUSSION THREAD

**Discussion by afifsyaifullah1512** [ivanwilliammd/satusehat-integration#123] [Ideas]
Selamat malam dok, apakah untuk encounter dan condition tidak ada Resourcetype Bundle?

**Comment by ivanwilliammd** [comment]
Untuk ini saat ini saya open for PR / need help untuk pengerjaan fitur ini...
Help wanted : @afifsyaifullah1512 @yudistirasd @YogiPristiawan @SyaefulKai 

Untuk per saat ini yang disupport adalah metode POST-nya saja, tetapi pembuatan array object bundle-nya masih manual

**[USULAN] :** 
Secara umum bundle itu pertama2 bisa diinisialisasi array-nya dahulu / backbone kosongan (resourceType Bundle, dll)
Kemudian saya usul bisa ada seperti inisialisasi Bundle

use ........\Bundle;

$encounter = new Encounter();
...... (diisi dengan object encounter)

$condition1 = new Condition()
...... (diisi dengan object condition1)

dst... (sejumlah berapa byk condition)

$bundle = new Bundle()
$bundle = $bundle->add($encounter)
$bundle = $bundle->add($condition1)
$bundle = $bundle->add($condition2)
$bundle->post()

**Comment by ivanwilliammd** [comment]
@afifsyaifullah1512 bisa dicoba v2.3.0.

Special thanks for @SyaefulKai for adding the bundle support...
Untuk dokumentasi di Wiki apakah boleh sekalian dibantu @SyaefulKai ?

Thanks

**Comment by SyaefulKai** [comment]
Boleh pak, untuk wikinya akan segera saya update

**Comment by afifsyaifullah1512** [comment]
oke dok, terimakasih updatenya, saya coba dulu

**Comment by SyaefulKai** [comment]
Tadi kodenya baru saya coba lagi.

Sepertinya ada bug yang harus diselesaikan dulu pak.

**Comment by ivanwilliammd** [comment]
Barusan saya sdh komen PR-nya @SyaefulKai, sepertinya perlu kembali dicek

Perlu dipahami link antara Encounter & Condition itu dua arah
Condition.encounter = $encounter_id
Encounter.diagnosis = array object berisi Conditionid & Conditiondisplay

https://github.com/ivanwilliammd/satusehat-integration/pull/27#pullrequestreview-1969909396

**Comment by afifsyaifullah1512** [comment]
$encounter->setSubject('{id_patient}', '{nama_pasien}'); // ID SATUSEHAT Pasien dan Nama SATUSEHAT

Dok, untuk mengisi bundle ini, kan harus ada ID Pasien (NIK) dan Nama pasien, apakah ini diambil dari Satusehat yang get_by_nik? Dan jika iya, result dari name nya kan disensor, apakah name yang di sensor itu yang di post ke  $encounter->setSubject('{id_patient}', '{nama_pasien}');

@ivanwilliammd

**Comment by ivanwilliammd** [comment]
1. Benar dari getByNik
2. Nama ini tdk disensor oleh server satusehat.. tapi kalau tdk bisa didapatkan --> bisa pakai nama yg tertera di medical record masing2

**Comment by ivanwilliammd** [comment]
Kudos untuk @SyaefulKai --> Fix updated di release v2.3.1
Untuk wiki juga sudah diupdate
